src/main/java/org/jenkinsci/plugins/displayurlapi/user/PreferredProviderUserProperty.java

```diff
@@ -33,22 +31,18 @@ public ProviderOption getProvider() {
     }
 
     public DisplayURLProvider getConfiguredProvider() {
-        return Iterables.find(DisplayURLProvider.all(), new Predicate<DisplayURLProvider>() {
-            @Override
-            public boolean apply(DisplayURLProvider input) {
-                return input.getClass().getName().equals(providerId);
-            }
-        }, null);
+        return DisplayURLProvider.all().stream()
+            .filter(input -> input.getClass().getName().equals(providerId))
+            .findFirst()
+            .orElse(null);
     }
 
     public List<ProviderOption> getAll() {
-        Iterable<ProviderOption> options = Iterables.transform(DisplayURLProvider.all(), new Function<DisplayURLProvider, ProviderOption>() {
-            @Override
-            public ProviderOption apply(DisplayURLProvider input) {
-                return new ProviderOption(input.getClass().getName(), input.getDisplayName());
-            }
-        });
-        return ImmutableList.copyOf(Iterables.concat(Lists.newArrayList(ProviderOption.DEFAULT_OPTION), options));
+        Stream<ProviderOption> defaultOption = Stream.of(ProviderOption.DEFAULT_OPTION);
+        Stream<ProviderOption> options = DisplayURLProvider.all().stream()
+            .map(input -> new ProviderOption(input.getClass().getName(), input.getDisplayName()));
+        Stream<ProviderOption> combined = Stream.concat(defaultOption, options);
+        return ImmutableList.copyOf(combined.collect(Collectors.toList()));
```

Comment:
Sometimes, as in this case, I find Java's stream API too verbose and harder to read than other solutions.
This could also be written as (untested):
```
List<ProviderOption> options = DisplayURLProvider.all().stream()
        .map(input -> new ProviderOption(input.getClass().getName(), input.getDisplayName()))
        .collect(Collectors.toList());
return ImmutableList.builder().add(ProviderOption.DEFAULT_OPTION).addAll(options).build();
```

Reply:
indeed that reads better :smile: I just don't use Guava that much.